client/signup/steps/domains/index.jsx

```diff
@@ -213,24 +208,29 @@ class DomainsStep extends React.Component {
 
 	handleSkip = () => {
 		const domainItem = undefined;
+
 		this.props.submitSignupStep( { stepName: this.props.stepName, domainItem }, { domainItem } );
 		this.props.goToNextStep();
 	};
 
-	submitWithDomain = googleAppsCartItem => {
+	submitWithDomain = ( googleAppsCartItem, shouldHideFreePlan = false ) => {
 		const suggestion = this.props.step.suggestion,
-			isPurchasingItem = Boolean( suggestion.product_slug ),
-			siteUrl = isPurchasingItem
-				? suggestion.domain_name
-				: suggestion.domain_name.replace( '.wordpress.com', '' ),
+			isPurchasingItem = suggestion && Boolean( suggestion.product_slug ),
+			siteUrl =
+				suggestion &&
+				( isPurchasingItem
+					? suggestion.domain_name
+					: suggestion.domain_name.replace( '.wordpress.com', '' ) ),
```

Comment:
This is minor style nit, but this set of declarations is starting to get pretty long. What do you think about breaking this up into separate `const` declarations for each of these (`suggestion`, `isPurchasingItem`, `siteUrl`, and `domainItem`?

Reply:
Addressed in [00f8cc2](https://github.com/Automattic/wp-calypso/pull/37554/commits/00f8cc226c263c8b4f42037c189f453ee734c181)